tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1399,6 +1399,12 @@ void SetBranchesHelper(TTree *inputTree, TTree &outputTree, const std::string &i
          leaf->GetLeafCount() ? std::string(leaf->GetLeafCount()->GetName()) : std::to_string(leaf->GetLenStatic());
       const auto btype = leaf->GetTypeName();
       const auto rootbtype = TypeName2ROOTTypeName(btype);
+      if (rootbtype == ' ') {
+         Warning("Snapshot",
+                 "RDataFrame::Snapshot: could not correctly construct a leaflist for C-style array in column %s. This "
+                 "column will not be written out.",
+                 bname);
+      }
       const auto leaflist = std::string(bname) + "[" + counterStr + "]/" + rootbtype;
```

Comment:
Since, it is possible that we reach this line and have `rootbtype` be `' '`, I think we should not add a type hint in this case.

So I would suggest having `const auto leaflist = std::string(bname) + "[" + counterStr + "]" + (rootbtype != ' ') ? ("\" + rootbtype) : "";` or something similar.

Reply:
as discussed we now skip the rest of the logic if the leaflist could not be constructed